made a better reload

MrBooks36 opened the pull request:
the /reload doesn't fix everything so i made a better one /trigger BugFix  recompile required tho

MrBooks36 committed:
Add files via upload

MrBooks36 committed:
Delete src/src directory

MrBooks36 committed:
Add files via upload

Big-Con-Gaming commented:
Ok, few things:
To my knowledge the only thing bugfix.mcfunction is teleporting the player to a fixed location inside the editor and then running reload, while also being accessible to /trigger. Here are my concerns:
Just checked, It isn't fully instanced actually, so no problems here. We should prob. look into that in the future.
2. Scheduling the function to run every 5 ticks is the same as placing the function inside of tick.mcfunction, but less efficient and runs less often.
3. I don't see the value in using /trigger, since we're giving the player admin perms at this stage in Alpha. This means they'll be able to run /reload on their own without the need for a trigger. This route seems to just add extra scoreboards without much use for them.

Big-Con-Gaming commented:
Ok update:
I now understand what this issue is trying to fix. If you kill all entities without killing the player, or other destructive bug-testing, /reload won't assign you a new lane if you are already inside a lane, so the lane won't be reconstructed if damaged. I think we could implement a method inside load.mcfunction to automatically regenerate all lanes on /reload though.

MrBooks36 committed:
Update bugfix.mcfunction

MrBooks36 committed:
Update tick.mcfunction

MrBooks36 committed:
Update tick.mcfunction

MrBooks36 committed:
Update load.mcfunction

MrBooks36 committed:
Update tick.mcfunction

MrBooks36 committed:
Update load.mcfunction

MrBooks36 committed:
Update tick.mcfunction

Connor-WWW committed:
Final Cleanup

Removed redundant code + cleaned up unused functions.

MrBooks36 commented:
done

Connor-WWW committed:
Final Fixes

Connor-WWW committed:
Retain position

Players inside the editor will now stay inside the editor, same with lane like usual.
